benchmark_v2/io_benchmark.py

```diff
@@ -0,0 +1,285 @@
+import os
+from glob import glob
+
+import pytest
+
+import arkouda as ak
+
+TYPES = ("int64", "float64", "uint64", "str")
+
+FILETYPES = ("HDF5", "PARQUET")
+
+COMPRESSIONS = ("snappy", "gzip", "brotli", "zstd", "lz4")
+
+
+def _write_files(a, ftype, dtype, compression=None):
+    for i in range(pytest.io_files):
+        a.to_hdf(f"{pytest.io_path}_hdf_{dtype}_{i:04}") if ftype == "HDF5" else a.to_parquet(
+            f"{pytest.io_path}_par_{compression}_{dtype}_{i:04}",
+            compression=compression if pytest.io_compressed else None,
+        )
+
+
+def _write_multi(a, dtype, compression=None):
+    for i in range(pytest.io_files):
+        a.to_parquet(
+            f"{pytest.io_path}_par_multi_{compression}_{dtype}_{i:04}",
+            compression=compression if pytest.io_compressed else None,
+        )
+
+
+def _append_files(a, dtype, compression):
+    for i in range(pytest.io_files):
+        mode = "truncate"  # First iteration of each trial needs to truncate the file
+        for key in a:
+            val = a[key]
+            val.to_parquet(
+                f"{pytest.io_path}_par_append_{compression}_{dtype}_{i:04}",
+                dataset=key,
+                mode=mode,
+                compression=compression if pytest.io_compressed else None,
+            )
+            mode = "append"

```

Comment:
The reason for the truncate is because of the option to only run the write benchmarks without deleting the files. The `truncate` is there in case the file already exists we aren't creating a "column name already exists in file" error due to multiple writes being performed without first deleting

Reply:
Ah 10-4 that makes sense.